pkg/downloader/chart_downloader.go

```diff
@@ -125,6 +137,34 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+// https://github.com/kubernetes/helm/issues/2937
+// tries to fix URL because of BUG #2937
+// transforms URL from
+// https://url.to.repo/path?key=value/chart-0.1.0.tgz
+// into
+// https://url.to.repo/path/chart-0-1-0.tgz?key=value
+func fixupURL(chartURL string) (string, error) {
+	queryStringStartIndex := strings.Index(chartURL, "?")
+
+	if queryStringStartIndex == -1 {
+		return "", errors.New("Could not fixup URL")
```

Comment:
Can this be a better error (if this function is kept)? How would someone know that a URL was broken and in need of fixing? And, what URL is causing that problem? I don't see how someone could take action based on this error to figure out how to correct the problem.

Reply:
removed